rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java

```diff
@@ -112,12 +116,123 @@ void setUp() {
+                bridgeMainNetConstants.getFederationConstants(),
+                federationStorageProvider,
+                rskExecutionBlock,
+                allActivations
+            );
+
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withProvider(bridgeStorageProvider)
+                .withEventLogger(bridgeEventLogger)
+                .withActivations(allActivations)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+        }
+
+        @Test
+        void processValidationFailure_whenSvpFundTxHashUnsigned_shouldLogValidationFailureAndClearValue() {
+            // arrange
+            Sha256Hash svpFundTxHashUnsigned = BitcoinTestUtils.createHash(1);
+            bridgeStorageProvider.setSvpFundTxHashUnsigned(svpFundTxHashUnsigned);
+
+            // act
+            bridgeSupport.processValidationFailure(proposedFederation);
+
+            // assert
+            assertLogCommitFederationFailed();
+            assertProposedFederationWasCleared();
+            assertFalse(bridgeStorageProvider.getSvpFundTxHashUnsigned().isPresent());
+        }
+
+        @Test
+        void processValidationFailure_whenSvpFundTxSigned_shouldLogValidationFailureAndClearValue() {
+            // arrange
+            BtcTransaction svpFundTx = mock(BtcTransaction.class);
+            bridgeStorageProvider.setSvpFundTxSigned(svpFundTx);
+
+            // act
+            bridgeSupport.processValidationFailure(proposedFederation);
+
+            // assert
+            assertLogCommitFederationFailed();
+            assertProposedFederationWasCleared();
+            assertFalse(bridgeStorageProvider.getSvpFundTxSigned().isPresent());
+        }
+
+        @Test
+        void processValidationFailure_whenSvpSpendTxWFS_shouldLogValidationFailureAndClearSpendTxValues() {
+            // arrange
+            Keccak256 svpSpendTxCreationHash = RskTestUtils.createHash(1);
+            BtcTransaction svpSpendTx = mock(BtcTransaction.class);
+            Map.Entry<Keccak256, BtcTransaction> svpSpendTxWFS = new AbstractMap.SimpleEntry<>(svpSpendTxCreationHash, svpSpendTx);
+            bridgeStorageProvider.setSvpSpendTxWaitingForSignatures(svpSpendTxWFS);
+
+            // act
+            bridgeSupport.processValidationFailure(proposedFederation);
+
+            // assert
+            assertLogCommitFederationFailed();
+            assertProposedFederationWasCleared();
+            assertFalse(bridgeStorageProvider.getSvpSpendTxWaitingForSignatures().isPresent());
+            assertFalse(bridgeStorageProvider.getSvpSpendTxHashUnsigned().isPresent());
+        }
+
+        @Test
+        void processValidationFailure_whenSvpSpendTxHashUnsigned_shouldLogValidationFailureAndClearValue() {
```

Comment:
Can a `SvpSpendTxHashUnsigned` be set without setting `SvpSpendTxWaitingForSignatures`? Is this a possibility? I see both values are always set together [here](https://github.com/rsksmart/rskj/blob/c2688c4e033bd961514b33e8cbbbf51e069958f7/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1151)

Reply:
when the svpSpendTx is fully signed, is removed from the map. Check [here](https://github.com/rsksmart/rskj/blob/c2688c4e033bd961514b33e8cbbbf51e069958f7/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1735). But the hash is not removed until the spend tx is registered.

Reply:
Understood. Thanks.